packages/api/amplify_api/lib/src/decorators/authorize_http_request.dart

```diff
@@ -0,0 +1,111 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:http/http.dart' as http;
+import 'package:meta/meta.dart';
+
+/// Transforms an HTTP request according to auth providers that match the endpoint
+/// configuration.
+@internal
+Future<http.BaseRequest> authorizeHttpRequest(http.BaseRequest request,
+    {required AWSApiConfig endpointConfig,
+    required AmplifyAuthProviderRepository authProviderRepo}) async {
+  if (!request.headers.containsKey(AWSHeaders.authorization)) {
```

Comment:
To help with readability, could this be a guard clause? Something like...

```
if (request.headers.containsKey(AWSHeaders.authorization)) {
    return request;
}

final authType = endpointConfig.authorizationType;
...
```

Reply:
yep, good suggestion! will do in next rev